src/Nethermind/Nethermind.Db.Rocks/Statistics/DbMetricsUpdater.cs

```diff
@@ -0,0 +1,142 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Nethermind.Db.Rocks.Config;
+using Nethermind.Logging;
+using RocksDbSharp;
+
+namespace Nethermind.Db.Rocks.Statistics
+{
+    public class DbMetricsUpdater
+    {
+        private readonly string _dbName;
+        private readonly DbOptions _dbOptions;
+        private readonly RocksDb _db;
+        private readonly IDbConfig _dbConfig;
+        private readonly ILogger _logger;
+        private Timer? _timer;
+
+        public DbMetricsUpdater(string dbName, DbOptions dbOptions, RocksDb db, IDbConfig dbConfig, ILogger logger)
+        {
+            _dbName = dbName;
+            _dbOptions = dbOptions;
+            _db = db;
+            _dbConfig = dbConfig;
+            _logger = logger;
+        }
+
+        public void StartUpdating()
+        {
+            var offsetInSec = _dbConfig.StatsDumpPeriodSec * 1.1;
+
+            _timer = new Timer(UpdateMetrics, null, TimeSpan.FromSeconds(offsetInSec), TimeSpan.FromSeconds(offsetInSec));
+        }
+
+        private void UpdateMetrics(object? state)
+        {
+            try
+            {
+                // It seems that currently there is no other option with .NET api to extract the compaction statistics than through the dumped string
+                var compactionStatsString = _db.GetProperty("rocksdb.stats");
+                ProcessCompactionStats(compactionStatsString);
+
+                if (_dbConfig.EnableDbStatistics)
+                {
+                    var dbStatsString = _dbOptions.GetStatisticsString();
+                    // Currently we don't extract any DB statistics but we can do it here
+                }
+            }
+            catch (Exception exc)
+            {
+                _logger.Error($"Error when updating metrics for {_dbName} database.", exc);
+                // Maybe we would like to stop the _timer here to avoid logging the same error all over again?
+            }
+        }
+
+        public void ProcessCompactionStats(string compactionStatsString)
+        {
+            if (!string.IsNullOrEmpty(compactionStatsString))
+            {
+                var stats = ExctractStatsPerLevel(compactionStatsString);
+                UpdateMetricsFromList(stats);
+
+                stats = ExctractIntervalCompaction(compactionStatsString);
+                UpdateMetricsFromList(stats);
+            }
+            else
+            {
+                _logger.Warn($"No RocksDB compaction stats available for {_dbName} databse.");
+            }
+        }
+
+        private void UpdateMetricsFromList(List<(string Name, long Value)> levelStats)
+        {
+            if(levelStats != null)
+            {
+                foreach (var stat in levelStats)
+                {
+                    Metrics.MetricsDict[$"Db{_dbName}{stat.Name}"] = stat.Value;
```

Comment:
{_dbName}_db to be consistent with other stats

Reply:
Changed